### PR TITLE
Rebase reorder conf

### PR DESCRIPTION
**Pull request by nicoTolly**
Allow choice order to be specified in config

### CODE REVIEW

File: src/explorer/bandit_arm.rs [hunk 1]
```diff
@@ -197,7 +198,11 @@ impl<'a, 'b> Store<'a> for Tree<'a, 'b> {
                 return None;
             }
             let cut: f64 = { *unwrap!(self.cut.read()) };
-            let state = unwrap!(self.shared_tree.write()).descend(context, cut);
+            let state = unwrap!(self.shared_tree.write()).descend(
+                &self.config.choice_ordering,
```
Comment (Elarnon) [inline src/explorer/bandit_arm.rs]: We should not do this in this PR, but we may want to consider creating a struct in which we can put "relevant descent information" (maybe `DescendConfig`?) for the context + cut + choice ordering + additional data we may need later.

File: src/explorer/choice.rs [hunk 2]
```diff
@@ -81,83 +111,89 @@ where
...
-                ChoiceGroup::DimMap =>
-                    Box::new(fun.static_dims().enumerate()
-                             .flat_map(move |(i, lhs)| {
-                                 fun.static_dims().take(i).flat_map(move |rhs| {
-                                     let mappings = space.domain().get_thread_mapping(lhs.id(), rhs.id());
-                                     gen_choice(mappings.list(), &|m| {
-                                         Action::ThreadMapping(lhs.id(), rhs.id(), m)
-                                     })
+
+pub fn list<'a>(
```
Comment (Elarnon) [inline src/explorer/choice.rs]: Please swap the arguments either here or in `list_from_conf` for consistency.

File: src/explorer/choice.rs [hunk 3]
```diff
@@ -81,83 +111,89 @@ where
...
 /// This function is to be either removed or reimplemented eventually. It is just a replacement for
 /// the previous list implementation (exposes the choices in the same order). Default should
 /// preferably be handled in config file
-pub fn list<'a>(space: &'a SearchSpace<'a>) -> impl Iterator<Item = Choice> + 'a {
-    DEFAULT_ORDERING.list(space)
+pub fn default_list<'a>(space: &'a SearchSpace<'a>) -> impl Iterator<Item = Choice> + 'a {
+    list(DEFAULT_ORDERING.iter().cloned(), space)
+}
+
+pub fn list_from_conf<'a, 'b: 'a>(
```
Comment (Elarnon) [inline src/explorer/choice.rs]: Let's call this `list_with_config` instead.  `XXX_from_YYY` consumes the `YYY` usually.

**Comment by nicoTolly** [comment]
bors retry

**Pull request merged**